// api/src/middleware/auth.js
import jwt from 'jsonwebtoken';

export function authMiddleware(req, res, next) {
    try {
        const authHeader = req.headers.authorization;

        if (!authHeader || !authHeader.startsWith('Bearer ')) {
            return res.status(401).json({
                success: false,
                error: 'Missing or invalid authorization header'
            });
        }

        const token = authHeader.split(' ')[1];
        
        // Decode token tanpa verifikasi signature secret terlebih dahulu agar fleksibel dengan PocketBase JWT
        const decoded = jwt.decode(token);

        if (!decoded) {
            return res.status(401).json({
                success: false,
                error: 'Token format tidak valid atau gagal didecode'
            });
        }

        // PocketBase menyimpan data id di properti 'id' atau di dalam 'model.id' / 'record.id'
        req.user = {
            id: decoded.id || decoded.model?.id || decoded.record?.id || decoded.sub,
            email: decoded.email || decoded.model?.email || decoded.record?.email,
            name: decoded.name || decoded.model?.name || decoded.record?.name
        };

        if (!req.user.id) {
            return res.status(401).json({
                success: false,
                error: 'Token valid tapi tidak mengandung ID Pengguna yang sah'
            });
        }

        next();
    } catch (error) {
        return res.status(401).json({
            success: false,
            error: 'Authentication failed',
            details: error.message
        });
    }
}
